### ISSUE TITLE
Add `IO.traverse`

### ISSUE THREAD

**Issue by jdegoes** [scalaz/scalaz-zio#11]
Uses `par` for parallel composition.

```scala
object IO {
  def traverse[E, A, B, M[X] <: TraversableOnce[X]](in: M[A])(fn: A => IO[E, B])(
    implicit cbf: CanBuildFrom[M[A], B, M[B]]): IO[E, M[B]]
}
```

**Comment by leandrob13** [comment]
@jdegoes I want to take a stab at it:

```scala
def traverse[E, A, B, M[X] <: TraversableOnce[X]](in: M[A])(fn: A => IO[E, B])(
    implicit cbf: CanBuildFrom[M[A], B, M[B]]): IO[E, M[B]] = {

    def loopList(l: List[A], b: mutable.Builder[B, M[B]]): IO[E, M[B]] = l match {
      case Nil => now(b.result())
      case l :: ls => fn(l).flatMap(a => loopList(ls, b += a))
    }
    loopList(in.toList, cbf(in))
  }
```

Is there a performance consideration for choosing an `Iterator` over an immutable `List`?

**Comment by ktonga** [comment]
Closed by #33

**Comment by jdegoes** [comment]
@leandrob13 Tail-recursive loop over `List` will be as fast as iteration, maybe faster.
